server/src/main/java/org/apache/cloudstack/diagnostics/RetrieveDiagnosticsServiceImpl.java

```diff
@@ -161,15 +161,31 @@ public void set_configParams(Map<String, Object> _configParams) {
     @Inject
     MessageBus messageBus;
 
-    DiagnosticsKey<String> IPTablesRemove = new DiagnosticsKey<String>(String.class, "IPtables.remove", "The IPtables rules to be removed", null, null);
+    ConfigKey<Long> RetrieveDiagnosticsTimeOut = new ConfigKey<Long>("Advanced", Long.class, "retrieveDiagnostics.retrieval.timeout", "3600",
+            "The timeout setting in seconds for the overall API call", true, ConfigKey.Scope.Global);
+    ConfigKey<Boolean> enabledGCollector = new ConfigKey<>("Advanced", Boolean.class, "retrieveDiagnostics.gc.enabled",
+            "true", "Garbage collection on/off switch (true|false", true, ConfigKey.Scope.Global);
+    ConfigKey<String> RetrieveDiagnosticsFilePath = new ConfigKey<String>("Advanced", String.class, "retrieveDiagnostics.filepath",
+            "/tmp", "File path to use on the management server for all temporary data. This allows CloudStack administrators to determine where best to place the files.", true, ConfigKey.Scope.Global);
+    ConfigKey<Float> RetrieveDiagnosticsDisableThreshold = new ConfigKey<Float>("Advanced", Float.class, "retrieveDiagnostics.disablethreshold", "0.95",
+            "The percentage disk space cut-off before API call will fail", true, ConfigKey.Scope.Global);
+
+    ConfigKey<Long> RetrieveDiagnosticsFileAge = new ConfigKey<Long>("Advanced", Long.class, "retrieveDiagnostics.max.fileage", "86400",
+            "The diagnostics file age in seconds before considered for garbage collection", true, ConfigKey.Scope.Global);
+
+    ConfigKey<Long> RetrieveDiagnosticsInterval = new ConfigKey<Long>("Advanced", Long.class, "retrieveDiagnostics.gc.interval", "86400",
+            "The interval between garbage collection executions in seconds", true, ConfigKey.Scope.Global);
+
+
+/*    DiagnosticsKey<String> IPTablesRemove = new DiagnosticsKey<String>(String.class, "IPtables.remove", "The IPtables rules to be removed", null, null);
```

Comment:
please remove code instead of commenting it out. we have a versioning system to retrieve old code.